src/setting_sbp_cb.c

```diff
@@ -61,28 +81,20 @@ static void setting_update_value(settings_t *ctx, uint8_t *msg, uint8_t len)
     return;
   }
 
-  if (setting_data->watchonly) {
+  if ((filter & UPDATE_FILTER_WATCHONLY) && setting_data->watchonly) {
```

Comment:
```suggestion
  if (update_filter_check(filter, UPDATE_FILTER_WATCHONLY) && setting_data->watchonly) {
```
just an illustration of the above suggestion